src/helpers/get-data.js

```diff
@@ -254,3 +234,47 @@ export const getSharedAgenciesList = async (name, user) => {
     return [...agencies, name];
   }
 };
+
+export const getVesselViewPageWithFilter = (vessel) => {
+  const filter = {};
+  if (vessel.permitNumber) {
+    filter["vessel.permitNumber"] = vessel.permitNumber;
+  }
+
+  const vesselName = vessel.name || vessel.vessel;
+
+  if (vesselName) {
+    filter["vessel.name"] = vesselName;
+  }
+
+  return VIEW_VESSEL_PAGE.replace(":filter", JSON.stringify(filter))
+}
+
+export const goVesselViewPage = (vessel) => {
+  history.push(getVesselViewPageWithFilter(vessel))
+};
+
+export const getCrewViewPageWithFilter = (crew) => {
+  const filter = {};
+  if (crew.isCaptain || crew.rank === "captain") {
+    if (crew.license) {
+      filter["captain.license"] = crew.license;
+    }
+    if (crew.name) {
+      filter["captain.name"] = crew.name;
+    }
+  } else {
+    if (crew.license) {
+      filter["crew.license"] = crew.license;
+    }
+    if (crew.name) {
+      filter["crew.name"] = crew.name;
+    }
+  }
+
+  return VIEW_CREW_PAGE.replace(":filter", JSON.stringify(filter))
+};
+
+export const goCrewViewPage = (crew) => {
+  history.push(getCrewViewPageWithFilter(crew))
+};
```

Comment:
this is from `goToPageWithFilter` but now using the shared method `getCrewViewPageWithFilter` to get the link,
and this one only takes care of going to the page